pkg/recipes/terraform/config/types.go

```diff
@@ -61,4 +61,14 @@ type TerraformDefinition struct {
 	// Backend defines where Terraform stores its state.
 	// https://developer.hashicorp.com/terraform/language/state
 	Backend map[string]interface{} `json:"backend"`
+
+	// RequiredProviders is the list of required Terraform providers.
+	RequiredProviders map[string]*RequiredProviderInfo `json:"required_providers,omitempty"`
```

Comment:
Is there a reason why `json:"required_providers,omitempty"` is not camel case?

Reply:
this is because in the tf.json file,  the serialized json output is expected to have the field name as 'required_providers'

Reply:
@lakshmimsft this question can come up for others reading this code as well, adding link to the relevant doc in the comment will help here.

Reply:
updated in comments